### PR TITLE
handling of incompatible published codecs

### PR DESCRIPTION
**Pull request by cloudwebrtc**
js: https://github.com/livekit/client-sdk-js/pull/911

### CODE REVIEW

File: lib/src/participant/local.dart [hunk 1]
```diff
@@ -250,6 +243,30 @@ class LocalParticipant extends Participant<LocalTrackPublication> {
 
     await track.start();
 
+    String? primaryCodecMime;
+    for (var codec in trackInfo.codecs) {
+      primaryCodecMime ??= codec.mimeType;
+    }
+
+    if (primaryCodecMime != null) {
```
Comment (davidzhao) [inline lib/src/participant/local.dart:251]: should only update if track is video

File: lib/src/utils.dart [hunk 2]
```diff
@@ -620,3 +620,11 @@ class ScalabilityMode {
     return 'L${spatial}T${temporal}${suffix ?? ''}';
   }
 }
+
+String mimeTypeToVideoCodecString(String mimeType) {
+  final codec = mimeType.split('/')[1].toLowerCase();
```
Comment (davidzhao) [inline lib/src/utils.dart:625]: should guard if mimeType isn't valid

**Review by davidzhao — COMMENTED** [review]
one

File: lib/src/participant/local.dart (hunk 1 above)
Comment (cloudwebrtc) [reply lib/src/participant/local.dart:251]: flutter SDK uses a separate function `publishVideoTrack` to publish track, so there should be no need to check `track is video`

Comment (davidzhao) [reply lib/src/participant/local.dart:251]: ah that makes sense. sorry I missed that.

**Review by davidzhao — APPROVED** [review]
awesome!

**Pull request merged**